clientconn.go

```diff
@@ -137,7 +151,16 @@ func (cc *ClientConn) resetTransport(closeTransport bool) error {
 		if err != nil {
 			// TODO(zhaoq): Record the error with glog.V.
 			closeTransport = false
-			time.Sleep(backoff(retries))
+			if neterr, ok := err.(net.Error); ok {
```

Comment:
This does not work -- "ok" is always false because the real type of err is transport.ConnectionError instead of net.Error. You need to add Temporary() and Timeout() to transport.ConnectionError to make this work. This needs to be properly tested.